js.go

```diff
@@ -1214,6 +1368,276 @@ type StreamState struct {
 	Consumers int       `json:"consumer_count"`
 }
 
+// UpdateStream updates a Stream.
+func (js *js) UpdateStream(cfg *StreamConfig) (*StreamInfo, error) {
+	if cfg == nil || cfg.Name == _EMPTY_ {
+		return nil, ErrStreamNameRequired
+	}
+
+	req, err := json.Marshal(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	usSubj := js.apiSubj(fmt.Sprintf(JSApiStreamUpdateT, cfg.Name))
+	r, err := js.nc.Request(usSubj, req, js.wait)
+	if err != nil {
+		return nil, err
+	}
+	var resp JSApiStreamInfoResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, errors.New(resp.Error.Description)
+	}
+	return resp.StreamInfo, nil
+}
+
+// JSAPIStreamDeleteResponse is the response for a Stream delete request.
+type JSAPIStreamDeleteResponse struct {
+	APIResponse
+	Success bool `json:"success,omitempty"`
+}
+
+// DeleteStream deletes a Stream.
+func (js *js) DeleteStream(name string) error {
+	if name == _EMPTY_ {
+		return ErrStreamNameRequired
+	}
+
+	dsSubj := js.apiSubj(fmt.Sprintf(JSApiStreamDeleteT, name))
+	r, err := js.nc.Request(dsSubj, nil, js.wait)
+	if err != nil {
+		return err
+	}
+	var resp JSAPIStreamDeleteResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return err
+	}
+	if resp.Error != nil {
+		return errors.New(resp.Error.Description)
+	}
+	return nil
+}
+
+// StreamSnapshotConfig contains options for snapshotting a Stream.
+type StreamSnapshotConfig struct {
+	// Subject to deliver the chunks to for the snapshot.
+	DeliverSubject string `json:"deliver_subject"`
+	// Do not include consumers in the snapshot.
+	NoConsumers bool `json:"no_consumers,omitempty"`
+	// Optional chunk size preference.
+	// Best to just let server select.
+	ChunkSize int `json:"chunk_size,omitempty"`
+	// Check all message's checksums prior to snapshot.
+	CheckMsgs bool `json:"jsck,omitempty"`
+	// Timeout is the amount of time to wait for the snapshot to stream to the
+	// client.
+	Timeout time.Duration `json:"-"`
```

Comment:
My first thought was to do just that, but then I realized that this is a config that we "share" with the server, and I was just worried that adding something here could possibly conflict later on with what is in the server. This is why I suggested to be passed as a parameter to SnaphotStream() API, the same way we have a timeout for sub's NextMsg() for instance.

But I am interested in other people thought here, namely, do people think it is ok to add the Timeout to the configuration, without exporting the JSON tag so that this is not sent to the server. Any chance that this becomes a conflict in the future? @derekcollison @ripienaar

Reply:
Aah, whoops. Missed that part. Makes sense to me. I changed the signature to

```
SnapshotStream(name string, timeout time.Duration, cfg *StreamSnapshotConfig
```

Reply:
@ripienaar Any opinion on the question of having the Timeout in the config (with a risk of a conflict later on if that were to be added in the server side stream snapshot config) or as an argument of the function?

Reply:
Personally I'd avoid it